### ISSUE TITLE
Flight upgrade doesn't let you initiate flight while in survival gamemode

### ISSUE THREAD

**Issue by DrAkashic** [Direwolf20-MC/JustDireThings#295]
### Just Dire Things version

1.4.9

### Minecraft Version

1.21.1

### (Neo)Forge Version

21.1.79

### Modpack & Version

A small custom pack. **(SEE UPDATE/COMMENT)**

### Do you have optifine or Rubidium installed?

None

### Describe the issue

The T4 creative flight upgrade for the chestplate doesn't allow the player to start creative flight.

### Steps to reproduce

1. Launch modded Minecraft instance
2. Create new superflat creative world with default settings
3. Grab portable generator, fuel canister, 5x Eclipse Ember blocks, Eclipse Ember Chestplate, Smithing Table, and Upgrade: Flight items
4. Put the Eclipse Ember blocks into the fuel canister, and the canister into the portable generator to begin charging the E.E. Chestplate
5. Place the Smithing table and use it to apply the Upgrade: Flight to the E.E. Chestplate
6. Equip the E.E. Chestplate
7. Switch gamemode to survival (Either via /gamemode survival or F3+F4)
8. Ensure that the Flight upgrade is enabled by opening the Tool Settings UI, checking that the upgrade is enabled, and it is set to 'Active if Equipped'.
9. Close all UIs and attempt to initiate flight by double-tapping the jump key, as with creative flight

### Expected behaviour

The player begins to fly, and the pocket generator burns fuel to keep up with the energy cost of the upgrade.

### Screenshots

![image](https://github.com/user-attachments/assets/15d523a4-a1e3-4380-a317-3e69a3f19a34)

### Additional information

Despite being unable to **start** flying, it is still possible to **maintain** creative flight using the Flight: Upgrade.

- Repeat repro steps 1 through 6
- Skip step 7 and continue through to completing step 9
- Regular vanilla creative flight will start (Note however that the pocket generator will begin to burn fuel as the E.E. Chestplate begins to consume RF energy from it's internal buffer)
- Follow step 7, switching to survival **while flying in the air in creative mode**
- The player continues to fly, even though they are now in survival, and the upgrade now seems to work as intended.
- The upgrade will stop working as intended (See original issue description) once the player has stopped flying by landing on the ground again

**Comment by DrAkashic** [comment]
Found the sneaky incompatibility!

I hunted through the differences between the pack I was using and the 1.21.1 DW20 pack half at a time until I found the mod that was causing the problem.
Turns out, something about Lithium's optimization rewrites causes the upgrade to stop working as described above.

Seeing as how this is a mod incompatibility issue and not a bug with JustDireThings itself, I'm going to close the ticket myself.
